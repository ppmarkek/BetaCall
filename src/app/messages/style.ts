import { Box, Button, Flex } from '@chakra-ui/react';
import styled from '@emotion/styled';

export const Wrapper = styled(Flex)`
  height: calc(100svh - 85px);
  min-height: 700px;
`;

export const LeftPanel = styled(Flex)`
  flex-direction: column;
  justify-content: space-between;
  padding: 20px 15px;
  width: 20%;
  border-right: 1px solid #eceef5;

  .active {
    background: #f8f9fc;

    svg {
      fill: #6b59cc;
    }

    .titleText {
      color: #6b59cc;
    }
  }
`;

export const BorderBox = styled(Box)`
  margin: 15px;
  border-bottom: 1px solid #eceef5;
`;

export const LeftPanelButton = styled(Button)`
  padding: 15px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  gap: 20px;
  width: 100%;
  height: 70px;
  background: #fff;
  border-radius: 6px;
  transition: all 0.3s;

  svg {
    fill: #8083a3;
    transition: fill 0.3s;
  }

  .titleText {
    color: #1a1c1d;
    transition: color 0.3s;
  }

  :hover {
    background: #f8f9fc;

    svg {
      fill: #6b59cc;
    }

    .titleText {
      color: #6b59cc;
    }
  }
`;
