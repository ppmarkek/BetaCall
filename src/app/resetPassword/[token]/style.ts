import { Flex } from '@chakra-ui/react';
import styled from '@emotion/styled';
import Link from 'next/link';

export const Wrapper = styled(Flex)`
  width: 420px;
  flex-direction: column;
  text-align: center;
  align-items: center;
  gap: 50px;
`;

export const StyledLink = styled(Link)`
  font-family: 'Lato', sans-serif;
  font-size: 16px;
  font-weight: 700;
  line-height: 19px;
  text-decoration: none;
  color: #6b59cc;
  cursor: pointer;
`;
