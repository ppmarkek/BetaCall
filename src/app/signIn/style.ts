import { Flex } from '@chakra-ui/react';
import styled from '@emotion/styled';
import Link from 'next/link';

export const Wrapper = styled(Flex)`
  width: 420px;
  flex-direction: column;
  text-align: center;
  gap: 50px;
`;

export const StyledCheckbox = styled.input`
  -webkit-appearance: none;
  background: #f0f0f3;
  appearance: none;
  border-radius: 50%;
  width: 20px;
  height: 20px;
  cursor: pointer;
  position: relative;
  outline: none;

  &:checked {
    background: #6b59cc;
  }

  &:checked::before {
    content: '';
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 10px;
    height: 10px;
    border-radius: 50%;
    background: #fff;
  }
`;

export const BoxOr = styled(Flex)`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;

  ::before,
  ::after {
    content: '';
    flex-grow: 1;
    height: 1px;
    background-color: #eeeeee;
  }

  ::before {
    margin-right: 12px;
    max-width: 100px;
  }

  ::after {
    margin-left: 12px;
    max-width: 100px;
  }
`;

export const StyledLink = styled(Link)`
  font-family: 'Lato', sans-serif;
  font-size: 16px;
  font-weight: 700;
  line-height: 19px;
  text-decoration: none;
  color: #6b59cc;
  cursor: pointer;
`;
