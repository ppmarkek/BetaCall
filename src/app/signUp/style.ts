import { Flex } from '@chakra-ui/react';
import styled from '@emotion/styled';

export const Wrapper = styled(Flex)`
  width: 420px;
  flex-direction: column;
  text-align: center;
  gap: 50px;
`;

export const StyledCheckbox = styled.input`
  -webkit-appearance: none;
  background: #f0f0f3;
  appearance: none;
  border-radius: 50%;
  width: 20px;
  height: 20px;
  cursor: pointer;
  position: relative;
  outline: none;

  &:checked {
    background: #6b59cc;
  }

  &:checked::before {
    content: '';
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 10px;
    height: 10px;
    border-radius: 50%;
    background: #fff;
  }
`;

export const BoxOr = styled(Flex)`
  width: 190px;
  align-items: center;
  justify-content: center;
  position: relative;

  ::before {
    content: '';
    width: 70px;
    position: absolute;
    top: 50%;
    left: 0;
    border-bottom: 1px solid #eeeeee;
  }

  ::after {
    content: '';
    width: 70px;
    position: absolute;
    top: 50%;
    right: 0;
    border-bottom: 1px solid #eeeeee;
  }
`;
