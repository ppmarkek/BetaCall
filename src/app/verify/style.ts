import styled from '@emotion/styled';

export const StyledLink = styled.a`
  font-family: 'Lato', sans-serif;
  font-size: 16px;
  font-weight: 700;
  line-height: 19px;
  text-decoration: none;
  color: #6b59cc;
  cursor: pointer;
`;
