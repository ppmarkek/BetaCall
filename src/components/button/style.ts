import { Button } from '@chakra-ui/react';
import styled from '@emotion/styled';

export const FilledButton = styled(Button)<{
  bgColor?: string;
  textColor?: string;
}>`
  padding: 12px 30px;
  border-radius: 10px;
  border: 1px solid ${({ bgColor }) => bgColor || '#6B59CC'};
  background: ${({ bgColor }) => bgColor || '#6B59CC'};
  color: ${({ textColor }) => textColor || '#fff'};
  transition: all 0.3s;
  cursor: pointer;
  :hover {
    background: unset;
    color: ${({ bgColor }) => bgColor || '#6B59CC'};
  }
`;

export const OutlineButton = styled(Button)<{
  bgColor?: string;
  textColor?: string;
}>`
  padding: 12px 30px;
  border-radius: 10px;
  border: 1px solid ${({ bgColor }) => bgColor || '#6B59CC'};
  background: unset;
  color: ${({ textColor }) => textColor || '#fff'};
  transition: all 0.3s;
  cursor: pointer;
  :hover {
    background: ${({ textColor }) => textColor || '#fff'};
    color: ${({ bgColor }) => bgColor || '#6B59CC'};
  }
`;
