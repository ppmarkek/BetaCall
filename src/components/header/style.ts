import { Box, Flex, PopoverContent } from '@chakra-ui/react';
import styled from '@emotion/styled';

export const LeftNavigatior = styled(Flex)<{ bigWidth?: boolean }>`
  position: absolute;
  height: 100svh;
  width: ${({ bigWidth }) => (bigWidth ? '270px' : '85px')};
  left: 0;
  flex-direction: column;
  justify-content: center;
  border-right: 1px solid #eceef5;
  min-height: 800px;
  overflow: hidden;
  transition: all 0.3s;
  .activeIcon {
    border-color: #eceef5;
  }
  .activeBigBox {
    background: #f8f9fc;
  }
`;

export const Navigatior = styled(Flex)`
  width: 100%;
  height: 85px;
  border-bottom: 1px solid #eceef5;
  justify-content: space-between;
  align-items: center;
  padding: 0px 30px;
`;

export const IconBox = styled(Flex)`
  position: absolute;
  gap: 22px;
  top: 20px;
  left: 22px;
`;

export const StyledIconBox = styled(Box)`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 40px;
  height: 40px;
  margin: 0px 22px;
  border: 1px solid #fff;
  border-radius: 8px;
  background: #fff;
  transition: all 0.3s;
`;

export const StyledIconBigBox = styled(Flex)`
  align-items: center;
  height: 70px;
  width: 270px;
  overflow: hidden;
  transition: all 0.3s ease;
  p {
    transition: all 0.3s ease;
  }
  :hover p {
    color: #1a1c1d;
  }
  :hover .iconBox {
    border-color: #eceef5;
  }
`;

export const MenuIcon = styled(Box)`
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  width: 40px;
  height: 40px;
  border: 1px solid #eceef5;
  border-radius: 8px;
  transition: all 0.3s;
  cursor: pointer;
  :hover {
    background: #eceef5;
  }
`;

export const StyledIconHeader = styled(Box)`
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  width: 40px;
  height: 40px;
  border: 1px solid #eceef5;
  border-radius: 8px;
  transition: all 0.3s;
  cursor: pointer;
  svg {
    fill: #8083a3;
    transition: all 0.3s;
  }

  :hover {
    background: #eceef5;
    svg {
      fill: #6b59cc;
    }
  }
`;

export const BorderBox = styled(Box)<{ bigMenu?: boolean }>`
  width: 15px;
  height: 2px;
  background: ${({ bigMenu }) => (bigMenu ? '#6B59CC' : '#8083a3')};
  ::after {
    content: ' ';
    position: absolute;
    top: 13px;
    width: 15px;
    height: 2px;
    transform: ${({ bigMenu }) => (bigMenu ? 'translateX(2px)' : 'none')};
    background: ${({ bigMenu }) => (bigMenu ? '#6B59CC' : '#8083a3')};
  }
  ::before {
    content: ' ';
    position: absolute;
    bottom: 13px;
    width: 15px;
    height: 2px;
    background: ${({ bigMenu }) => (bigMenu ? '#6B59CC' : '#8083a3')};
    transform: ${({ bigMenu }) => (bigMenu ? 'translateX(-2px)' : 'none')};
  }
`;

export const SearchIcon = styled(Box)`
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  width: 40px;
  height: 40px;
  border: 1px solid #eceef5;
  border-radius: 8px;
  transition: all 0.3s;
  cursor: pointer;
  :hover {
    background: #eceef5;
  }
`;

export const SearchBarContainer = styled(Flex)<{ $active: boolean }>`
  position: absolute;
  right: 0;
  border: 1px solid #eceef5;
  background: ${({ $active }) => ($active ? '#ECEEF5' : '#fff')};
  border-radius: 9px;
  padding-left: ${({ $active }) => ($active ? '20px' : '0')};
  transition:
    width 0.3s ease,
    padding-left 0.3s ease,
    background-color 0.3s ease;
  width: ${({ $active }) => ($active ? '570px' : '40px')};
  height: 40px;
  cursor: pointer;

  .icon-container {
    position: absolute;
    width: 20px;
    height: 20px;
    right: 9px;
    top: 50%;
    transform: translateY(-50%);
  }

  input {
    width: calc(100% - 40px);
    display: ${({ $active }) => ($active ? 'block' : 'none')};
    border: none;
    outline: none;
    background: transparent;
    font-size: 14px;
    color: #1a1c1d;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    ::placeholder {
      color: #1a1c1d;
    }
  }
`;

export const StyledPopoverContent = styled(PopoverContent)`
  position: absolute;
  padding: 10px;
  right: 0;
  top: 50px;
  width: auto;
`;

export const MessagesHeader = styled(Flex)`
  padding-left: 30px;
  justify-content: space-between;
  width: calc(80% - 80px);
`;

export const BorderPadding = styled(Box)`
  margin: 0px 20px;
  border-right: 1px solid #eceef5;
  height: 25px;
`;
