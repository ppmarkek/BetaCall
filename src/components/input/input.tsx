import { Icon } from '@chakra-ui/react';
import Typography from '../typography/typography';
import { InputBox, StyledInput } from './style';
import { IconType } from 'react-icons';
import { Tooltip } from '../ui/tooltip';

type TextInputProps = Omit<
  React.InputHTMLAttributes<HTMLInputElement>,
  'size'
> & {
  title?: string;
  iconElement?: IconType;
  placeholder?: string;
  error?: boolean;
  errorText?: string;
};

export default function TextInput({
  title,
  placeholder,
  iconElement,
  error,
  errorText,
  ...rest
}: TextInputProps) {
  const IconComponent = iconElement;

  return (
    <InputBox>
      <Typography variant="Button" color={'#8083A3'}>
        {title}
      </Typography>
      <Tooltip
        showArrow
        content={errorText}
        contentProps={{
          css: {
            '--tooltip-bg': '#ff808b',
            padding: '10px',
            display: error ? 'block' : 'none',
          },
        }}
        openDelay={500}
        closeDelay={100}
      >
        <StyledInput
          {...rest}
          $icon={!!IconComponent}
          placeholder={placeholder}
          className={error ? 'input-error' : ''}
          color={error ? '#ff808b' : '#1a1c1d'}
        />
      </Tooltip>
      {IconComponent && (
        <Icon
          className={`input-icon ${error ? 'input-icon-error' : ''}`}
          position={'absolute'}
          right={0}
          bottom={'8px'}
          fontSize={'20px'}
        >
          <IconComponent />
        </Icon>
      )}
    </InputBox>
  );
}
