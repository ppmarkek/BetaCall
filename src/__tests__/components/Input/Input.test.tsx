import React from 'react';
import { render, screen } from '../../../../test-utils';
import '@testing-library/jest-dom';
import TextInput from '@/components/input/input';
import userEvent from '@testing-library/user-event';
import { LuTestTube } from 'react-icons/lu';
import { StyledInput } from '@/components/input/style';

describe('Text Input Component', () => {
  it('renders the text input with title', () => {
    render(<TextInput title="Input test" />);
    expect(screen.getByText('Input test')).toBeInTheDocument();
  });

  it('renders the input with error state', () => {
    render(<TextInput error={true} errorText="Invalid input" />);
    const input = screen.getByRole('textbox');
    expect(input).toHaveClass('input-error');
  });

  it('renders the input with error and icon (without onIconClick)', () => {
    render(<TextInput error={true} iconElement={LuTestTube} />);
    const icon = screen.getByTestId('icon-textInput');
    expect(icon).toHaveClass('input-icon', 'input-icon-error');
  });

  it('renders the input with icon when error is false', () => {
    render(<TextInput iconElement={LuTestTube} error={false} />);
    const icon = screen.getByTestId('icon-textInput');
    expect(icon).toHaveClass('input-icon');
    expect(icon).not.toHaveClass('input-icon-error');
  });

  it('calls onIconClick when icon is clicked', async () => {
    const handleClick = jest.fn();
    render(
      <TextInput
        error={true}
        iconElement={LuTestTube}
        onIconClick={handleClick}
      />
    );
    const icon = screen.getByTestId('icon-textInput');
    await userEvent.click(icon);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('matches the snapshot', () => {
    const { container } = render(<TextInput />);
    expect(container).toMatchSnapshot();
  });
});

describe('Styled Components (style.ts)', () => {
  it('matches snapshot with default color', () => {
    const { container } = render(<StyledInput />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with custom color', () => {
    const { container } = render(<StyledInput color={'#000'} />);
    expect(container).toMatchSnapshot();
  });
});
