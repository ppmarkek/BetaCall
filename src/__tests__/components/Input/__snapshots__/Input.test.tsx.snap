// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Components (style.ts) matches snapshot with custom color 1`] = `
.emotion-0 {
  color: #000;
  padding: 20px 0px;
  padding-left: 0px;
  padding-right: 0px;
  border: unset;
  border-radius: unset;
  border-bottom: 1px solid #eceef5;
  color: #000;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

@layer recipes {
  .emotion-0 {
    width: 100%;
    min-width: var(--input-height);
    outline: 0;
    position: relative;
    -webkit-appearance: none;
    -moz-appearance: none;
    -ms-appearance: none;
    appearance: none;
    text-align: start;
    border-radius: var(--chakra-radii-l2);
    height: var(--input-height);
    --focus-color: var(--chakra-colors-color-palette-focus-ring);
    --error-color: var(--chakra-colors-border-error);
    font-size: var(--chakra-font-sizes-sm);
    line-height: 1.25rem;
    padding-inline: var(--chakra-spacing-3);
    --input-height: var(--chakra-sizes-10);
    background: var(--chakra-colors-transparent);
    --bg-currentcolor: var(--chakra-colors-transparent);
    border-width: 1px;
    border-color: var(--chakra-colors-border);
    --focus-ring-color: var(--chakra-colors-color-palette-focus-ring);
  }

  .emotion-0:is(:disabled, [disabled], [data-disabled], [aria-disabled=true]) {
    opacity: 0.5;
    cursor: not-allowed;
  }

  .emotion-0:is([data-invalid], [aria-invalid=true], [data-state=invalid]) {
    --focus-ring-color: var(--error-color);
    border-color: var(--error-color);
  }

  .emotion-0:is(:focus-visible, [data-focus-visible]) {
    outline-offset: 0px;
    outline-width: var(--focus-ring-width, 1px);
    outline-color: var(--focus-ring-color);
    outline-style: var(--focus-ring-style, solid);
    border-color: var(--focus-ring-color);
  }
}

.emotion-0:focus-visible {
  outline: unset;
  border-bottom: 1px solid #000;
}

<div>
  <input
    class="chakra-input emotion-0"
  />
</div>
`;

exports[`Styled Components (style.ts) matches snapshot with default color 1`] = `
.emotion-0 {
  padding: 20px 0px;
  padding-left: 0px;
  padding-right: 0px;
  border: unset;
  border-radius: unset;
  border-bottom: 1px solid #eceef5;
  color: #1a1c1d;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

@layer recipes {
  .emotion-0 {
    width: 100%;
    min-width: var(--input-height);
    outline: 0;
    position: relative;
    -webkit-appearance: none;
    -moz-appearance: none;
    -ms-appearance: none;
    appearance: none;
    text-align: start;
    border-radius: var(--chakra-radii-l2);
    height: var(--input-height);
    --focus-color: var(--chakra-colors-color-palette-focus-ring);
    --error-color: var(--chakra-colors-border-error);
    font-size: var(--chakra-font-sizes-sm);
    line-height: 1.25rem;
    padding-inline: var(--chakra-spacing-3);
    --input-height: var(--chakra-sizes-10);
    background: var(--chakra-colors-transparent);
    --bg-currentcolor: var(--chakra-colors-transparent);
    border-width: 1px;
    border-color: var(--chakra-colors-border);
    --focus-ring-color: var(--chakra-colors-color-palette-focus-ring);
  }

  .emotion-0:is(:disabled, [disabled], [data-disabled], [aria-disabled=true]) {
    opacity: 0.5;
    cursor: not-allowed;
  }

  .emotion-0:is([data-invalid], [aria-invalid=true], [data-state=invalid]) {
    --focus-ring-color: var(--error-color);
    border-color: var(--error-color);
  }

  .emotion-0:is(:focus-visible, [data-focus-visible]) {
    outline-offset: 0px;
    outline-width: var(--focus-ring-width, 1px);
    outline-color: var(--focus-ring-color);
    outline-style: var(--focus-ring-style, solid);
    border-color: var(--focus-ring-color);
  }
}

.emotion-0:focus-visible {
  outline: unset;
  border-bottom: 1px solid #000;
}

<div>
  <input
    class="chakra-input emotion-0"
  />
</div>
`;

exports[`Text Input Component matches the snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 10px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  position: relative;
}

.emotion-0 .input-icon {
  fill: #8083a3;
  -webkit-transition: fill 0.3s;
  transition: fill 0.3s;
}

.emotion-0:focus-within .input-icon:not(.input-icon-error) {
  fill: #1a1c1d;
}

.emotion-0 .input-icon-error {
  fill: #ff808b;
}

.emotion-1 {
  color: #8083A3;
  font-family: Lato;
  font-size: 16px;
  font-weight: 400;
  line-height: 19px;
}

.emotion-2 {
  color: #1a1c1d;
  padding: 20px 0px;
  padding-left: 0px;
  padding-right: 0px;
  border: unset;
  border-radius: unset;
  border-bottom: 1px solid #eceef5;
  color: #1a1c1d;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

@layer recipes {
  .emotion-2 {
    width: 100%;
    min-width: var(--input-height);
    outline: 0;
    position: relative;
    -webkit-appearance: none;
    -moz-appearance: none;
    -ms-appearance: none;
    appearance: none;
    text-align: start;
    border-radius: var(--chakra-radii-l2);
    height: var(--input-height);
    --focus-color: var(--chakra-colors-color-palette-focus-ring);
    --error-color: var(--chakra-colors-border-error);
    font-size: var(--chakra-font-sizes-sm);
    line-height: 1.25rem;
    padding-inline: var(--chakra-spacing-3);
    --input-height: var(--chakra-sizes-10);
    background: var(--chakra-colors-transparent);
    --bg-currentcolor: var(--chakra-colors-transparent);
    border-width: 1px;
    border-color: var(--chakra-colors-border);
    --focus-ring-color: var(--chakra-colors-color-palette-focus-ring);
  }

  .emotion-2:is(:disabled, [disabled], [data-disabled], [aria-disabled=true]) {
    opacity: 0.5;
    cursor: not-allowed;
  }

  .emotion-2:is([data-invalid], [aria-invalid=true], [data-state=invalid]) {
    --focus-ring-color: var(--error-color);
    border-color: var(--error-color);
  }

  .emotion-2:is(:focus-visible, [data-focus-visible]) {
    outline-offset: 0px;
    outline-width: var(--focus-ring-width, 1px);
    outline-color: var(--focus-ring-color);
    outline-style: var(--focus-ring-style, solid);
    border-color: var(--focus-ring-color);
  }
}

.emotion-2:focus-visible {
  outline: unset;
  border-bottom: 1px solid #000;
}

<div>
  <div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    />
    <input
      class="chakra-input chakra-tooltip__trigger emotion-2"
      data-part="trigger"
      data-scope="tooltip"
      data-state="closed"
      dir="ltr"
      id="tooltip::r5::trigger"
    />
  </div>
</div>
`;
